packages/medusa/src/api/routes/store/carts/calculate-taxes.ts

```diff
@@ -34,91 +34,98 @@ export default async (req, res) => {
 
   const headerKey = req.get("Idempotency-Key") || ""
 
-  let idempotencyKey: IdempotencyKey
-  try {
-    idempotencyKey = await idempotencyKeyService.initializeRequest(
-      headerKey,
-      req.method,
-      req.params,
-      req.path
-    )
-  } catch (error) {
-    console.log(error)
-    res.status(409).send("Failed to create idempotency key")
-    return
-  }
+    const manager: EntityManager = req.scope.resolve("manager")
+  await manager.transaction(async (transactionManager) => {
+    let idempotencyKey: IdempotencyKey
+    try {
+      idempotencyKey = await idempotencyKeyService.withTransaction(transactionManager).initializeRequest(
+        headerKey,
+        req.method,
+        req.params,
+        req.path
+      )
+    } catch (error) {
+      console.log(error)
+      res.status(409).send("Failed to create idempotency key")
+      return
```

Comment:
todo: the initialization of the idempotency key must happen within a transaction otherwise the subsequent calls to `idempotencyKeyService.workStage` will create nested transactions which I don't think will work.

Reply:
@srindom I did pass the transaction, what do you mean?